### PR TITLE
implement multiindex.swaplevel 

### PR DESCRIPTION
**Pull request by RainFung**
implement multiindex.swaplevel by rename sdf column

### CODE REVIEW

**Comment by RainFung** [comment]
Hello @HyukjinKwon . Can you tell me why using `F.struct` to combining columns in MultiIndex `init` constructor?
```
    def __init__(self, kdf: DataFrame):
        assert len(kdf._internal._index_map) > 1
        scol = F.struct(kdf._internal.index_scols)
        data_columns = kdf._sdf.select(scol).columns
        internal = kdf._internal.copy(scol=scol,
                                      column_index=[(col, None) for col in data_columns],
                                      column_index_names=None)
        IndexOpsMixin.__init__(self, internal, kdf)
```

**Comment by RainFung** [comment]
Passed Python 3.7 Travis CI Test.

**Comment by HyukjinKwon** [comment]
> Hello @HyukjinKwon . Can you tell me why using F.struct to combining columns in MultiIndex init constructor?

I think this's because Multiindex has multiple columns for each level. InternalFrame was designed to have a single column `scol`. So, having a struct column was added to match and fix into this design so far

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -831,6 +831,55 @@ def name(self) -> str:
...
+        index_columns = self._internal.index_columns
+        sdf = self._internal._sdf
+        before = index_columns[i]
+        after = index_columns[j]
+        sdf = sdf.withColumnRenamed(after, 'tmp')\
+            .withColumnRenamed(before, after)\
+            .withColumnRenamed('tmp', before)
+        result = MultiIndex(DataFrame(_InternalFrame(sdf=sdf,
+                                                     index_map=[(i, None) for i in index_columns])))
+        return result
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py]:
```
Expected:
    MultiIndex([('x', 'a', 1),
                ('y', 'b', 2)],
               )
Got:
    MultiIndex(levels=[['x', 'y'], ['a', 'b'], [1, 2]],
               labels=[[0, 1], [0, 1], [0, 1]])
```

Can you fix the test failures at other versions too?

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -831,6 +831,55 @@ def name(self) -> str:
...
+        Returns
+        -------
+        MultiIndex
+            A new MultiIndex.
+
+        See Also
+        --------
+        Series.swaplevel : Swap levels i and j in a MultiIndex.
+        Dataframe.swaplevel : Swap levels i and j in a MultiIndex on a
+            particular axis.
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: We don't have these yet?

File: databricks/koalas/tests/test_indexes.py [hunk 3]
```diff
@@ -319,3 +319,10 @@ def test_multiindex_from_arrays(self):
         kidx = ks.MultiIndex.from_arrays(arrays)
 
         self.assert_eq(pidx, kidx)
+
+    def test_multiindex_swaplevel(self):
+        pidx = pd.DataFrame({'a': ['a', 'b']}, index=[['a', 'b'], ['x', 'y'], [1, 2]]).index
+        kidx = ks.DataFrame({'a': ['a', 'b']}, index=[['a', 'b'], ['x', 'y'], [1, 2]]).index
+
+        if LooseVersion(pd.__version__) >= LooseVersion("0.25.3"):
+            self.assert_eq(pidx.swaplevel(0, 1), kidx.swaplevel(0, 1))
```
Comment (ueshin) [inline databricks/koalas/tests/test_indexes.py]: Could you add tests to use index names?

**Comment by RainFung** [comment]
move to https://github.com/databricks/koalas/pull/1104

**Pull request closed without merging**